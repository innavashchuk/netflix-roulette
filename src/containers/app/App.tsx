import * as React from 'react';
import Footer from '../../components/footer/Footer';
import Header from '../header/Header';
import SearchHeader from '../search-header/Search-Header';
import ErrorBoundary from '../../components/error-boundary/Error-Boundary';
import Main from '../main/Main';
import './app.scss';

export interface AppState {
  searchValue: string
}

class App extends React.Component<Record<string, unknown>, AppState> {
  constructor(props: Record<string, unknown>) {
    super(props);
    this.state = {
      searchValue: ''
    }
  }

  render(): React.ReactElement {
    return (
      <div className="app-container">
        <ErrorBoundary>
        <div className="app-container__inner">
          <Header>
            <SearchHeader />
          </Header>
          <Main searchValue={this.state.searchValue} />
        </div>
        </ErrorBoundary>
        <Footer />
      </div>
    );
  }
};

export default App;
