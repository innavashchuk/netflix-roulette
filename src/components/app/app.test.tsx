import * as React from 'react';
import {render} from '@testing-library/react';
import App from './app';

jest.mock('../search-movies-page/search-movies-page', () => () => <div>StartPage</div>);
jest.mock('../movie-details/movie-details', () => () => <div>MovieDetails</div>);
jest.mock('../custom-alert/custom-alert', () => () => <div>CustomAlert</div>);

describe('App', () => {
    it('should initialize', () => {
        const {asFragment} = render(<App/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
