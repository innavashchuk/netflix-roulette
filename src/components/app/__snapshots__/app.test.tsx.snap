// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should initialize 1`] = `
<DocumentFragment>
  <div>
    CustomAlert
  </div>
  <div>
    CustomLoader
  </div>
  <div>
    StartPage
  </div>
</DocumentFragment>
`;
